keylime/cert_utils.py

```diff
@@ -42,8 +42,7 @@ def x509_der_cert(der_cert_data: bytes) -> Certificate:
     """
     try:
         return x509.load_der_x509_certificate(data=der_cert_data, backend=default_backend())
-    except Exception as err:
-        logger.warning("Failed to parse DER data with python-cryptography: %s", err)
```

Comment:
OK, fair explanation. I created an issue to track this: #1559

I think the message should inform the user trying to configure Keylime which certificates are problematic so that they can decide whether to accept them or not. For the CA certificates, this would require the user to remove the malformed certificates from the trusted certificate store (and we probably should inform them about this).

For the CA certificates, the warning should appear once during initialization. For other certificates, I think it is still important to log that they are problematic. Probably the best is to allow the user to configure whether they want to ignore and accept this or not. When the user chooses to ignore this, then the messages should be suppressed.

Reply:
@ansasaki If you could have a look at the fix I've implemented for #1559 in the latest commit, that'd be great. The code that generates the warning is on [lines 108-159 of registrar_agent.py](https://github.com/keylime/keylime/blob/7527175fdfecbe0859eb9155f41bea6c6ee7410e/keylime/models/registrar/registrar_agent.py#L108-L159). This generates a warning only when one or more certificates are added or updated in an agent record and lets the user know which certificates are at fault (in a single warning message). A new config option has been added to optionally suppress this warning or reject all problematic certs (the default is just to warn).

It looks like CodeQL is upset about these new log messages because it thinks I am logging a certificate (which is not the case). I think I'll try to just change the name of the relevant variable and see if that fixes it (let me know if you have a better idea).

I also added a template to migrate the config to include this new option. Let me know if I did this correctly or not. I ran the "convert_config" script and it didn't complain.

One last thing: Since I needed to create a new config template anyway, I used this opportunity to change the default logging format to include the request ID. When the X-Request-Id header is set in a request, its value is appended to the all log messages produced during the handling of that request. This makes it easier to see which log messages relate to which request. However, users will not notice a change until the agent is updated to set this header.

Reply:
Okay, so changing that variable name did not fix the CodeQL warning. I assume it needs to be suppressed by a maintainer.

Reply:
I realised that `cert_utils.x509_der_cert` and `cert_utils.x509_pem_cert` functions are still called from other places (e.g., indirectly from the tenant) in the codebase and my fix in [commit 7527175](https://github.com/keylime/keylime/pull/1523/commits/7527175fdfecbe0859eb9155f41bea6c6ee7410e) only applies when certs are added to an agent record in the registrar. So, I've added back the warning in `cert_utils` and made copies of these functions without the warning in `models.types.Certificate`. That way, the improved warning is used when possible, and the old warning is used otherwise.